### PR TITLE
Adding status bar and controller, fixing zoom box grabbing undo

### PR DESCRIPTION
**Pull request by kotoole1**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27940593%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27940618%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27944133%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27944134%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27976902%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27977141%22%2C%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23issuecomment-90939828%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%2026b47729407da26e782d64740692c9a8c2ddb6e6%20Paranormal/ParanormalTests/WindowControllerTests.swift%2049%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27940618%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Whoops.%20Fixing%20now.%22%2C%20%22created_at%22%3A%20%222015-04-08T03%3A42%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2672344%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kotoole1%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-04-08T05%3A34%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2672344%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kotoole1%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/ParanormalTests/WindowControllerTests.swift%3AL32-61%22%7D%2C%20%22Pull%2026b47729407da26e782d64740692c9a8c2ddb6e6%20Paranormal/Paranormal/StatusBar/StatusBarViewController.swift%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27940593%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22These%20are%20moved%20with%20essentially%20no%20changes%20from%20the%20windowController%22%2C%20%22created_at%22%3A%20%222015-04-08T03%3A41%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2672344%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kotoole1%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-04-08T05%3A34%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2672344%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kotoole1%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/Paranormal/StatusBar/StatusBarViewController.swift%3AL1-48%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23issuecomment-90939828%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Aship%3A%20Just%20that%20one%20small%20thing%22%2C%20%22created_at%22%3A%20%222015-04-08T14%3A54%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20386d135e95d2c24e75ad4753c0cc30295584a712%20Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27977141%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Thank%20you%22%2C%20%22created_at%22%3A%20%222015-04-08T14%3A54%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift%3AL26-33%22%7D%2C%20%22Pull%20386d135e95d2c24e75ad4753c0cc30295584a712%20Paranormal/Paranormal/StatusBar/StatusBarViewController.swift%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/lukemetz/Paranormal/pull/111%23discussion_r27976902%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Remove%20the%20%3F%20at%20the%20end%20of%20this.%20%22%2C%20%22created_at%22%3A%20%222015-04-08T14%3A52%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1248454%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukemetz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Paranormal/Paranormal/StatusBar/StatusBarViewController.swift%3AL1-49%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 386d135e95d2c24e75ad4753c0cc30295584a712 Paranormal/Paranormal/StatusBar/StatusBarViewController.swift 40'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/111#discussion_r27976902'>File: Paranormal/Paranormal/StatusBar/StatusBarViewController.swift:L1-49</a></b>
- <a href='https://github.com/lukemetz'></a> Remove the ? at the end of this.
- [ ] <a href='#crh-comment-Pull 386d135e95d2c24e75ad4753c0cc30295584a712 Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift 5'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/111#discussion_r27977141'>File: Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift:L26-33</a></b>
- <a href='https://github.com/lukemetz'></a> Thank you
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/111#issuecomment-90939828'>General Comment</a></b>
- <a href='https://github.com/lukemetz'></a> :ship: Just that one small thing
- [x] <a href='#crh-comment-Pull 26b47729407da26e782d64740692c9a8c2ddb6e6 Paranormal/Paranormal/StatusBar/StatusBarViewController.swift 8'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/111#discussion_r27940593'>File: Paranormal/Paranormal/StatusBar/StatusBarViewController.swift:L1-48</a></b>
- <a href='https://github.com/kotoole1'></a> These are moved with essentially no changes from the windowController
- [x] <a href='#crh-comment-Pull 26b47729407da26e782d64740692c9a8c2ddb6e6 Paranormal/ParanormalTests/WindowControllerTests.swift 49'></a> &nbsp;<b><a href='https://github.com/lukemetz/Paranormal/pull/111#discussion_r27940618'>File: Paranormal/ParanormalTests/WindowControllerTests.swift:L32-61</a></b>
- <a href='https://github.com/kotoole1'></a> Whoops. Fixing now.

<a href='https://www.codereviewhub.com/lukemetz/Paranormal/pull/111?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/lukemetz/Paranormal/pull/111?approve=1'></a>&nbsp;<a href='https://github.com/lukemetz/Paranormal/pull/111'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: Paranormal/Paranormal/StatusBar/StatusBarViewController.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+public let PNNotificationZoomChanged = "PNNotificationZoomChanged"
+
+public class StatusBarViewController : PNViewController {
+
+    @IBOutlet weak var zoomTextField: NSTextField!
+
```
Comment (kotoole1) [inline Paranormal/Paranormal/StatusBar/StatusBarViewController.swift]: These are moved with essentially no changes from the windowController

File: Paranormal/ParanormalTests/WindowControllerTests.swift [hunk 2]
```diff
@@ -32,29 +32,29 @@ class WindowControllerTests : QuickSpec {
...
+//                                ranNotification = true
+//                        })
+//
+//                        windowController.zoomSetFromGUI(windowController.zoomField)
+//                        let date = NSDate(timeIntervalSinceNow: 0.1)
+//                        NSRunLoop.currentRunLoop().runUntilDate(date)
+//                        expect(ranNotification).toEventually(beTrue(()))
+//                    }
+//                }
+//            }
```
Comment (kotoole1) [inline Paranormal/ParanormalTests/WindowControllerTests.swift]: Whoops. Fixing now.

Comment (kotoole1) [inline Paranormal/ParanormalTests/WindowControllerTests.swift]: :+1:<a href='#crh-update-none'></a>

File: Paranormal/Paranormal/StatusBar/StatusBarViewController.swift (hunk 1 above)
Comment (kotoole1) [inline Paranormal/Paranormal/StatusBar/StatusBarViewController.swift]: :+1:<a href='#crh-update-complete'></a>

File: Paranormal/Paranormal/StatusBar/StatusBarViewController.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    @IBAction public func setEditorViewMode(sender: NSSegmentedControl) {
+        if let newViewMode = EditorViewMode(rawValue: sender.selectedSegment) {
+            document?.editorViewMode = newViewMode
+        } else {
+            log.error("editor mode set to unknown value \(sender.selectedSegment)")
+        }
+    }
+
+    @IBAction public func lightToggled(sender: NSButton) {
+        let panelsViewController = document?.singleWindowController?.panelsViewController?
```
Comment (lukemetz) [inline Paranormal/Paranormal/StatusBar/StatusBarViewController.swift]: Remove the ? at the end of this.

File: Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift [hunk 4]
```diff
@@ -26,7 +26,7 @@ class ToolsViewControllerTests: QuickSpec {
                 for button in buttons {
                     if (button == currentbutton) {
                         expect(button?.bordered).to(beTrue())
-                    }else{
+                    } else {
```
Comment (lukemetz) [inline Paranormal/ParanormalTests/Tools/ToolsViewControllerTests.swift]: Thank you

**Comment by lukemetz** [comment]
:ship: Just that one small thing

**Pull request merged**